### ISSUE TITLE
Formulary organization

### ISSUE THREAD

**Issue by pheuer** [PlasmaPy/PlasmaPy#1097]
The formulary namespace is getting pretty cluttered, especially `formulary.parameters` which is basically a catch-all category with 38 functions (including aliases) and counting.

I think we should consider taking a cue from the NRL formulary and separating some of these functions into categories based on their units, eg. "frequencies", "lengths", and "velocities". Some other quantities could be easily refactored into this scheme: for example all of the functions in `drifts.py` would fit nicely into "velocities".  Other candidate categories might be "pressures", "densities", and "diffusivities". The functions in `dimensionless.py` could also be made into a "dimensionless" category at the same level. I guess I'm proposing putting these at the same level as `parameters` is now, eg. `formulary.frequencies.py`. I would replace `paramters.py` with `misc.py` in this new scheme.

This would also make it easier to ensure that functions are in the right place. For example, right now `Hall_parameter` is in `parameters` when it definitely belongs in `dimensionless.py`. 

One con is that not all functions make sense in this scheme: for example `collision_frequency` belongs in `collisions` more than it belongs with `gyrofrequency` . But, I'd argue that distinction is no more arbitrary than leaving `collision_frequency` out of `parameters`. 

Finally, I think the number of aliases is cluttering the documentation. ~~Is there a way to hide these in `autodoc` on the module level so that they only appear listed as "aliases" when you click on the individual function?~~ It looks like some approach grouping the aliases together will work better.

**Comment by rocco8773** [comment]
I absolutely agree the `formulary` is starting to get cluttered and could use a little reorganizing.  However, there's no non-trivial way to improve this.  The mechanisms (i.e. `__all__`) used for organizing the package namespace is also used for controlling the documentation.  So, restricting the `formulary` namespace can really hinder the exposure of our functionality.   Additionally, if we want good exposure of functionality on the `formulary` namespace then it will always have a LOT of names.  This is kind of like the `astropy.units` namespace.  It will always be large because a lot of functionality needs to be exposed there.

**Comment by pheuer** [comment]
I don't know much about Sphinx or automodapi, but maybe[ this `:skip:` command](https://sphinx-automodapi.readthedocs.io/en/latest/automodapi.html#module-sphinx_automodapi.automodapi) could be used to hide functions from the docs? If that hides it from the list of module functions but still allows it to be set as an alias, that would be a solution?

Alternately, since the aliases really are one-to-one replacements, they could be hidden and then just mentioned in the docstring of the function they replace?

**Comment by rocco8773** [comment]
The problem with introducing something `:skip:` is that it has to be manually done, which means we're moving away from automating documentation to non-trivially increasing our manual documentation.  I'm not saying we can't figure something out, but I don't see `:skip:` as being a fruitful route.

First we all need to agree on how we want the `formulary` to organized and how we want the namespace to be exposed.  Once there's an agreement on that, I think we can do some trickery with programmatically defining `__all__` in the `__init__.py` and all sub-modules of the `formulary`.

**Comment by StanczakDominik** [comment]
That gave me ideas on what to do with all here... I'll report back with an incremental improvement.

**Comment by StanczakDominik** [comment]
#1099 cleans things up a little. https://plasmapy--1099.org.readthedocs.build/en/1099/formulary/parameters.html is an example of how I envision it working, once I get sphinx to detect stuff in the submodule.

This keeps the alias implementation in the same file, just dynamically provides a separate namespace from which to import it WHILE removing it from the main file's `__all__` list, thus increasing readability of the documentation list of functions.

Alternatively, I could play around with https://stackoverflow.com/a/28345836/4417567. If you can star import from a Namespace, we could probably completely skip the additional file part.

**Comment by pheuer** [comment]
I agree that #1099 could be a good solution for the aliases part of this problem! I'd propose doing as well as splitting `paramters.py` as described above. Together I think those would close this issue.

**Comment by StanczakDominik** [comment]
At the same time, thinking in terms of the implementation - would it not be preferable if we could just split the Functions list [in the docs, e.g. here](https://plasmapy--1099.org.readthedocs.build/en/1099/formulary/parameters.html) into a few sub-list based on...

you know, return annotations already encode what kind of quantity the output is:

```python
>>> unit =  u.m / u.s
>>> unit
Unit("m / s")
>>> unit.physical_type
'speed'
```

We could potentially use that, somehow.

**Comment by pheuer** [comment]
That would certainly work well for the documentation. It doesn't address clutter in the code itself though: `parameters.py` is almost 2000 lines and growing. I don't in principle have a problem with all of these being in a "parameters" module if the docs are clearly divided and if somehow the code is organized to make it easier scroll through.

**Comment by StanczakDominik** [comment]
Fair! I didn't notice it growing this large 🥲

**Comment by namurphy** [comment]
Closed by #1434, #1439, #1444, #1446, #1448, and #1453.
